docs/docsite/rst/dev_guide/developing_modules_general.rst

```diff
@@ -266,20 +278,111 @@ Key parts include always importing the boilerplate code from
 
         from ansible.module_utils.basic import *
 
-And instantiating the module class like:
+
+main() and AnsibleModule & argument_spec
+++++++++++++++++++++++++++++++++++++++++
+
+
+# FIXME Flesh out example showing all options
+
+# FIXME Add a module & integration test that defends all of this (I think modules can go relative to the roles dir)
 
 .. code-block:: python
 
     def main():
         module = AnsibleModule(
-            argument_spec = dict(
-                state     = dict(default='present', choices=['present', 'absent']),
-                name      = dict(required=True),
-                enabled   = dict(required=True, type='bool'),
-                something = dict(aliases=['whatever'])
-            )
+            argument_spec=dict(
+                state=dict(default='present', choices=['present', 'absent']),
```

Comment:
Should probably be `required=True` - my understanding is that a default state is now frowned upon? (If not we should get that clarified once and for all)